conda_build/metadata.py

```diff
@@ -458,6 +464,17 @@ def build_id(self):
                         res.append(s + v.strip('*'))
                     break
 
+        for ms in self.ms_depends():
+            if ms.name in config.versions:
+                try:
+                    v = ms.spec.split()[1]
+                except IndexError:
+                    break
+                if any(i in v for i in ',|>!<'):
+                    break
+                res.append(ms.name + v.strip('*'))
+
```

Comment:
The `build_id()` (and thus the build string) will now mention any packages that have a specified `version`.  But in this implementation, the *order* of the package names in the build string will depend on the order in which they happened to be listed in `meta.yaml`.  Instead -- just for consistent naming conventions -- should they be sorted before they get added to the `build_id`?